boards/arm/mimxrt1050_evk/mimxrt1050_evk.dts

```diff
@@ -28,11 +28,6 @@
 	};
 
 	chosen {
-#if defined(CONFIG_CODE_ITCM)
-		zephyr,flash = &itcm0;
-#elif defined(CONFIG_CODE_HYPERFLASH)
-		zephyr,flash = &hyperflash0;
```

Comment:
Wouldn't it be cleaner if the selection was handled by a board .overlay file? E.g. `mimxrt1050_evk_itcm.overlay`, `mimxrt1050_evk_hyperflash.overlay`. We would keep usage of `zephyr,flash` property consistent and FLASH_BASE_ADDRESS could still be defined in DTS. The selection of board overlay file could be done in CMakeLists.txt based on Kconfig option.

Reply:
The problem than is you've made it so that the full dts isn't know until after Kconfig runs, which means we can't utilize the .dts in things like sanitycheck or Kconfig itself.

Reply:
Of course, sorry. That's not going to work.